### DISCUSSION TITLE
Cumulative Counter

### DISCUSSION THREAD

**Discussion by srstrong** [open-telemetry/opentelemetry-erlang#582] [Q&A]
Hi,

I have a Counter and have with the default of delta temporality, I see the counter reseting on each collection, which makes sense to me.  However, if I set my reader to have cumulative temporality I was expecting to see the counter not get reset, but  I'm observing is the the counter is the sum of the last two checkpoints whereas I was expecting it to just increase forever.  Am I misunderstanding what cumulative temporality is supposed to indicate?  (I understand the risk of cardinality explosion etc - in my system, that's not a risk)

**Comment by tsloughter** [comment]
Yes, it should increase forever. Summing the last two checkpoints should increase forever, right? Or are you seeing a reset in there too?

If you are using the hex package of `experimental` that may be the issue. I'll be publishing new versions once https://github.com/open-telemetry/opentelemetry-erlang/pull/581 is merged. The metrics implementation has changed a lot but is now "stable".

**Comment by srstrong** [comment]
I'm running on head from github - from what I can see in otel_aggregation_sum,  `previous_checkpoint` is set to `checkpoint`,  `checkpoint` is set to `value`, and `value` is reset to zero.  On collect, the value returned is then `checkpoint + previousCheckpoint`.  I suspect `previousCheckpoint` should be getting set to `previousCheckpoint + checkpoint`
